lib/auth0/api/authentication_endpoints.rb

```diff
@@ -1,22 +1,193 @@
+
+      # Logins using phone number/verification code.
+      # @see https://auth0.com/docs/auth-api#!#post--ro_with_sms
+      # @param phone_number [string] User's phone number.
+      # @param code [string] Verification code.
+      # @return [json] Returns the access token and id token
+      def phone_login(phone_number, code, scope = 'openid')
+        fail Auth0::InvalidParameter, 'Must supply a valid phone number' if phone_number.to_s.empty?
+        fail Auth0::InvalidParameter, 'Must supply a valid code' if code.to_s.empty?
+        request_params = {
+          client_id:  @client_id,
+          username:   phone_number,
+          password:   code,
+          scope:      scope,
+          connection: 'sms',
+          grant_type: 'password'
+        }
+        post('/oauth/ro', request_params)
+      end
+
+      # Retrives the SAML 2.0 metadata
+      # @see https://auth0.com/docs/auth-api#!#get--samlp--client_id-
+      # @param client_id [string] Client id
+      # @return [xml] SAML 2.0 metadata
+      def saml_metadata(client_id)
+        fail Auth0::InvalidParameter, 'Must supply a valid client_id' if client_id.to_s.empty?
+        get("/samlp/metadata/#{client_id}")
+      end
+
+      # Retrives the WS-Federation metadata
+      # @see https://auth0.com/docs/auth-api#!#get--wsfed--client_id-
+      # @return [xml] Federation Metadata
+      def wsfed_metadata
+        get('/wsfed/FederationMetadata/2007-06/FederationMetadata.xml')
+      end
+
+      # Validates a JSON Web Token (signature and expiration)
+      # @see https://auth0.com/docs/auth-api#!#post--tokeninfo
+      # @param id_token [string] Token's id.
+      # @return User information associated with the user id (sub property) of the token.
+      def token_info(id_token)
+        fail Auth0::InvalidParameter, 'Must supply a valid id_token' if id_token.to_s.empty?
+        request_params = { id_token: id_token }
+        post('/tokeninfo', request_params)
       end
 
-      # {https://auth0.com/docs/auth-api#!#post--delegation}
+      # Refreshes a delegation token
+      # @see https://auth0.com/docs/auth-api#!#post--delegation
+      # @param refresh_token [string] Token to refresh
+      # @param target [string] Target to sign the new token.
+      # @param scope [string] Defaults to openid. Can be 'openid name email'.
+      # @param api_type [string] Defaults to app. Can be aws, azure_sb, azure_blob, firebase, layer, salesforce_api,
+      #  salesforce_sandbox_api, sap_api or wams
+      # @param extra_parameters [hash] Extra parameters.
+      # @return [json] Returns the refreshed delegation token
+      def refresh_delegation(refresh_token, target, scope = 'openid', api_type = 'app', extra_parameters = {})
+        fail Auth0::InvalidParameter, 'Must supply a valid token to refresh' if refresh_token.to_s.empty?
+        request_params = {
+          client_id:      @client_id,
+          grant_type:     'urn:ietf:params:oauth:grant-type:jwt-bearer',
```

Comment:
is this constant used anywhere else? We could create a constant as we did with authentication